dev-util/gn/patches/gn-01463.patchset

```diff
@@ -0,0 +1,440 @@
+ 
+   @staticmethod
+   def known_platforms():
+-    return ['linux', 'darwin', 'msvc', 'aix', 'fuchsia']
++    return ['linux', 'darwin', 'msvc', 'aix', 'fuchsia', 'haiku']
+ 
+   def platform(self):
+     return self._platform
+@@ -66,9 +68,12 @@ class Platform(object):
+ 
+   def is_aix(self):
+     return self._platform == 'aix'
++  
++  def is_haiku(self):
++    return self._platform == 'haiku'
+ 
+   def is_posix(self):
+-    return self._platform in ['linux', 'darwin', 'aix']
++    return self._platform in ['linux', 'darwin', 'aix', 'haiku']
+ 
+ 
+ def main(argv):
+@@ -120,24 +125,15 @@ def main(argv):
+ 
+ 
+ def GenerateLastCommitPosition(host, header):
+-  ROOT_TAG = 'initial-commit'
+-  describe_output = subprocess.check_output(
+-      ['git', 'describe', 'HEAD', '--match', ROOT_TAG], shell=host.is_windows(),
+-      cwd=REPO_ROOT)
+-  mo = re.match(ROOT_TAG + '-(\d+)-g([0-9a-f]+)', describe_output)
+-  if not mo:
+-    raise ValueError(
+-        'Unexpected output from git describe when generating version header')
+-
+   contents = '''// Generated by build/gen.py.
+ 
+-#ifndef OUT_LAST_COMMIT_POSITION_H_
+-#define OUT_LAST_COMMIT_POSITION_H_
+-
+-#define LAST_COMMIT_POSITION "%s (%s)"
+-
+-#endif  // OUT_LAST_COMMIT_POSITION_H_
+-''' % (mo.group(1), mo.group(2))
++	#ifndef OUT_LAST_COMMIT_POSITION_H_
++	#define OUT_LAST_COMMIT_POSITION_H_
++	
++	#define LAST_COMMIT_POSITION "%s (%s)"
++	
++	#endif  // OUT_LAST_COMMIT_POSITION_H_
++	''' % (999999, "091169b")
+ 
+   # Only write/touch this file if the commit position has changed.
+   old_contents = ''
+@@ -221,6 +217,7 @@ def WriteGenericNinja(path, static_libraries, executables,
+       'darwin': 'build_mac.ninja.template',
+       'linux': 'build_linux.ninja.template',
+       'aix': 'build_aix.ninja.template',
++      'haiku': 'build_haiku.ninja.template',
+   }[platform.platform()])
+ 
+   with open(template_filename) as f:
+@@ -355,7 +352,7 @@ def WriteGNNinja(path, platform, host, options, linux_sysroot):
+     cflags.extend([
+         '-D_FILE_OFFSET_BITS=64',
+         '-D__STDC_CONSTANT_MACROS', '-D__STDC_FORMAT_MACROS',
+-        '-pthread',
++        '-pthread' if not platform.is_haiku() else '',
+         '-pipe',
+         '-fno-exceptions',
+         '-fno-rtti',
+@@ -384,6 +381,9 @@ def WriteGNNinja(path, platform, host, options, linux_sysroot):
+     elif platform.is_aix():
+       cflags_cc.append('-maix64')
+       ldflags.extend(['-maix64', '-pthread'])
++    if platform.is_haiku():
++      cflags_cc.append('-fPIC')
++      cflags_cc.extend(['-D_BSD_SOURCE', '-D__USE_XOPEN2K8'])
+ 
+     if options.use_lto:
+       cflags.extend(['-flto', '-fwhole-program-vtables'])
+@@ -424,6 +424,21 @@ def WriteGNNinja(path, platform, host, options, linux_sysroot):
+ 
+     ldflags.extend(['/DEBUG', '/MACHINE:x64'])
+ 
++    if platform.is_haiku():
++      libs.extend(['-lnetwork'])
++      libs.extend(['-lroot'])
++      libs.extend(['-lbsd'])
++      libs.extend(['-lstdc++'])
```

Comment:
Shouldn't this be taken care of by the C++ compiler?

Reply:
root will be, but the others will not be linked automatically, including stdc++.

Reply:
I tested gn this time without these flags and it seems to link fine. So the extra flags have been removed.